timesketch/lib/analyzers/geoip.py

```diff
@@ -0,0 +1,311 @@
+# Copyright 2021 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sketch analyzer plugin for geolocating IP addresses."""
+from __future__ import unicode_literals
+
+import os
+import ipaddress
+import logging
+
+from collections import defaultdict
+from typing import Tuple, Union
+
+from flask import current_app
+
+import geoip2.database
+import geoip2.errors
+import geoip2.webservice
+
+from timesketch.lib import emojis
+from timesketch.lib.analyzers import interface
+from timesketch.lib.analyzers import manager
+
+
+logger = logging.getLogger('timesketch.analyzers.geoip')
+
+
+class GeoIpClientAdapter(object):
+    """Base adapter interface for a third party geolocation service."""
```

Comment:
This is a great design, very clean context manager, nice!

Reply:
+1, I like it.